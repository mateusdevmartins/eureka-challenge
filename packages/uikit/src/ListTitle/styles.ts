import styled from 'styled-components';

export const MainView = styled.div`
  width: 100%;
  justify-content: space-between;
  display: flex;
  flex-direction: row;
  align-items: center;
  background-color: rgba(255, 255, 255, 1);
`;
export const ListItemContainer = styled.div`
  display: flex;
  flex-direction: row;
  align-items: center;
`;

export const ColumnTitle = styled.h1`
  font-style: normal;
  font-weight: 600;
  font-size: 13px;
  line-height: 17px;
  align-items: center;
  color: rgba(138, 148, 166, 1);
  padding: 0px 6px 0px 6px;
`;
