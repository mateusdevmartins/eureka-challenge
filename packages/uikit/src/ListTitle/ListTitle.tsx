import { MainView, ListItemContainer, ColumnTitle } from './styles';

export interface ListTitleProps {
  position: string;
  name: string;
  points: string;
}

export function ListTitle({ position, name, points }: ListTitleProps) {
  return (
    <MainView>
      <ListItemContainer>
        <ColumnTitle style={{ width: '24px' }}>{position}</ColumnTitle>
        <ColumnTitle>{name}</ColumnTitle>
      </ListItemContainer>
      <ColumnTitle>{points}</ColumnTitle>
    </MainView>
  );
}
