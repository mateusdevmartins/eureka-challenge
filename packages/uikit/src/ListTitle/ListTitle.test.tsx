import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { ListTitle } from './ListTitle';

describe('ListTitle ', () => {
  const props = {
    points: 'Pontos',
    position: 'Pos',
    name: 'Mateus',
  };

  it('should match snapshot', () => {
    const component = renderer.create(<ListTitle {...props} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('behavior', () => {
    const components = render(<ListTitle {...props} />);
    const { getByText } = components;
    expect(getByText(props.name)).toBeTruthy();
    expect(getByText(props.position)).toBeTruthy();
    expect(getByText(props.points)).toBeTruthy();
  });
});
