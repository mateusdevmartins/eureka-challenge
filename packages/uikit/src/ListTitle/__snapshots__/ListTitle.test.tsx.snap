// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListTitle  should match snapshot 1`] = `
<div
  className="styles__MainView-sc-9awpl6-0 fbRiBg"
>
  <div
    className="styles__ListItemContainer-sc-9awpl6-1 ftIQND"
  >
    <h1
      className="styles__ColumnTitle-sc-9awpl6-2 hfbWlz"
      style={
        Object {
          "width": "24px",
        }
      }
    >
      Pos
    </h1>
    <h1
      className="styles__ColumnTitle-sc-9awpl6-2 hfbWlz"
    >
      Mateus
    </h1>
  </div>
  <h1
    className="styles__ColumnTitle-sc-9awpl6-2 hfbWlz"
  >
    Pontos
  </h1>
</div>
`;
